pkg/cmd/infra/deployer/deployer.go

```diff
@@ -38,7 +47,20 @@ func NewCommandDeployer(name string) *cobra.Command {
 		Short: "Run the OpenShift deployer",
 		Long:  longCommandDesc,
 		Run: func(c *cobra.Command, args []string) {
-			if err := deploy(cfg); err != nil {
+			kClient, _, err := cfg.Config.Clients()
+			if err != nil {
+				glog.Fatal(err)
+			}
+
+			if len(cfg.DeploymentName) == 0 {
+				glog.Fatal("deployment is required")
+			}
+
+			if len(cfg.Namespace) == 0 {
```

Comment:
> there's a bunch of infrastructure that already does namespace defaulting

The namespace part of the cli context work hasn't be rebased into origin yet, it only made it in with https://github.com/GoogleCloudPlatform/kubernetes/pull/3470.  This can be updated next week after the rebase.